File: creational/singleton/singleton.ts
class Singleton {
    private static instance: Singleton;

    private constructor() {
    }

    public class getInstance(): Singleton {
        if(!Singleton.instance) {
            Singleton.instance = new Singleton()
        }
        return Singleton.instance
    }
}

function test() {
    const s1 = Singleton.getInstance();
    const s2 = Singleton.getInstance();

    if(s1 === s2) {
        console.log('Singleton work')
    } else {
        console.log('Singleton Faild')
    }
}

File: creational/singleton/singleton.py
class Singleton(object):
    __instance = None

    def __new__(cls, *args, **kwargs):
        if cls.__instance is None:
            cls.__instance = object.__new__(cls, *args, **kwargs)
        return cls.__instance


singleton1 = Singleton()
singleton2 = Singleton()
print(singleton1, singleton2)
